src/Adapter/Order/CommandHandler/AddCartRuleToOrderHandler.php

```diff
@@ -69,6 +77,30 @@ public function handle(AddCartRuleToOrderCommand $command): void
         $this->assertPercentCartRule($command);
         $order = $this->getOrder($command->getOrderId());
 
+        $this->contextStateManager
+            ->setCurrency(new Currency($order->id_currency))
+            ->setCustomer(new Customer($order->id_customer));
+
+        try {
+            $this->addCartRuleAndUpdateOrder($command, $order);
+        } catch (Exception $exception) {
```

Comment:
Catching exception to throw it again is not required And it also misses a case when no exception the context is not cleared Use try/finally instead the finally block is called wether an exception is caught or not, but it doesn't block the exception